js/type.js

```diff
@@ -1,590 +1,537 @@
-				
-				if (styleEles[i].innerHTML) stylesheets.push(Array.prototype.slice.call(styleEles)[i]); // add on page CSS
-			
-			}
-			
-		}
-			
-		this.sheets = stylesheets;
-		
-		this.loadStyles(0);
-	
-	},
-	
-	runSheets : function () {
-		
-		for (sheet in this.sheetTexts) {
-			for (property in this.properties) {
-				
-				var sheetext = this.sheetTexts[sheet],
-						func = this.properties[property];
-				
-				this.findRules(property, sheetext, func);
-			
-			}
-		}
-		
-	},
-	
-	checkRule : function (property) {
-		
-		var propertyCamel = property.replace(/(^|-)([a-z])/g, function (m1, m2, m3) { return m3.toUpperCase(); });
-		
-		if (('Webkit' + propertyCamel) in document.body.style 
-		 || ('Moz' + propertyCamel) in document.body.style 
-		 || ('O' + propertyCamel) in document.body.style 
-		 || property in document.body.style) return true;
-	
-	},
-	
-	findRules : function (property, sheettext, func) {
-		
-		var rules = { support: false };
-			
-		if (sheettext) {
-			
-			if (!this.checkRule(property)) { // check if rule is valid now
-			
-				var selreg = new RegExp('([^}{]+){([^}]+)?' + property.replace('-', '\\-') + '[\\s\\t]*:[\\s\\t]*([^;]+)', 'gi'), 
-						selmatch, i = 0;
-				
-				while (selmatch = selreg.exec(sheettext)) {
-		   		
-		   		var sel = selmatch[1].replace(/^([\s\n\r\t]+|\/\*.*?\*\/)+/, '').replace(/[\s\n\r\t]+$/, ''),
-		   				val = selmatch[3];
-		   		
-					rules['rule' + i] = {
-						
-						selector: sel,
-						property: property,
-						value: val
-						
-					};
-					
-					if (!stylefill.allRules[sel]) stylefill.allRules[sel] = new Object();
-					stylefill.allRules[sel][property] = val;
-					
-					i++;
-				
-				}
-			    
-			}
-			else rules.support = true;
-			
-			func(rules);
-		
-		}
-		
-	}
 
+    req.open("GET", url, true);
+
+    req.onreadystatechange = function () {
+
+      if (this.readyState === 4 && this.status === 200) {
+        stylefill.sheetTexts[url] = this.responseText;
+        stylefill.loadStyles(count + 1);
+      }
+      else if (this.readyState === 4 && this.status === 000) {
+        stylefill.loadStyles(count + 1);
+      }
+    };
+
+    req.send(null);
+  },
+
+  getStyleSheets: function (property, func) {
+
+    var sheetstext = [],
+      stylesheets = Array.prototype.slice.call(document.querySelectorAll('link[href*=".css"]')), // grab stylesheet links - not used yet
+      styleEles = document.getElementsByTagName('style');
+
+    if (styleEles.length > 0) {
+      for (var i in styleEles) {
+        if (styleEles[i].innerHTML) {
+          // add on page CSS
+          stylesheets.push(Array.prototype.slice.call(styleEles)[i]);
+        }
+      }
+    }
+
+    this.sheets = stylesheets;
+    this.loadStyles(0);
+  },
+
+  runSheets: function () {
+
+    for (var sheet in this.sheetTexts) {
+      for (var property in this.properties) {
+
+        if (this.sheetTexts.hasOwnProperty(sheet) && this.properties.hasOwnProperty(property)) {
```

Comment:
When using a for...in loop over objects, you should use hasOwnProperty to get only the keys that are on the object itself and not inherited ones.